src/decoder/tracking_player.cc

```diff
@@ -31,22 +33,28 @@ SerializedFrame TrackingPlayer::serialize_next( void )
   UncompressedChunk decompressed_frame = decoder_.decompress_frame( compressed_frame );
   if ( decompressed_frame.key_frame() ) {
     KeyFrame frame = decoder_.parse_frame<KeyFrame>( decompressed_frame );
+    SerializedFrame serialized_frame = decode_and_serialize( frame, source, compressed_frame );
 
-    return decode_and_serialize( frame, source, compressed_frame );
-  } else {
+    return make_pair( serialized_frame, FrameInfo( raw_data.offset, raw_data.length,
```

Comment:
I decided to keep them apart, since the frame info is just about hashes and offsets, but the serialized frame is about the bytes data and decompressed raster. I think it's cleaner this way.